src/main/java/com/github/princesslana/eriscasper/rest/guild/CreateGuildRoleRequest.java

```diff
@@ -0,0 +1,30 @@
+package com.github.princesslana.eriscasper.rest.guild;
+
+import java.util.Optional;
+import org.immutables.value.Value;
+
+/**
+ * @see <a
+ *     href="https://discordapp.com/developers/docs/resources/guild#create-guild-role-json-params">
+ *     https://discordapp.com/developers/docs/resources/guild#create-guild-role-json-params</a>
+ */
+@Value.Immutable
+public interface CreateGuildRoleRequest {
+  Optional<String> getName();
+
+  Optional<Integer> getPermissions();
+
+  Optional<Integer> getColor();
+
+  Optional<Boolean> isHoist();
+
+  Optional<Boolean> isMentionable();
+
+  static CreateGuildRoleRequest newRole() {
+    return ImmutableCreateGuildRoleRequest.builder().build();
+  }
+
+  static CreateGuildRoleRequest of(String name) {
```

Comment:
Is naming this `ofName` more consistent with other `of` methods we have?

Reply:
Not sure, we kind of switch, `of` not `ofDays` `ofNick` not `of`, so I mean it kind of goes back and forth.

Reply:
We should decide on a format and be consistent with it, it's mostly my fault for switching in between, just didn't know of the right format to use.

Reply:
I thought we might.

For me I think we should stick with `ofXxx` going forward, because using just `of` breaks down if there's multiple of's I might want that take a String.

Reply:
Alright, I agree, should I change all past `of`s, or just from now on be consistent with ofXxx.

Reply:
Nah, we don't need to change past ones yet. Let's be consistent going forward, and if we're woking on a class or functionality that has an `of`, we can change it then.

Reply:
Alright, sounds good, I'll change all of GuildRoute's `of`s just to keep this PR consistent and think about it going forward.